backend/src/libs/packages/socket/socket.service.ts

```diff
@@ -1,27 +1,47 @@
 import { type FastifyInstance } from 'fastify/types/instance';
 import { Server as SocketServer } from 'socket.io';
 
+import { type GeolocationCacheService } from '~/libs/packages/geolocation-cache/geolocation-cache.package';
 import { logger } from '~/libs/packages/logger/logger.js';
 
 import { ServerSocketEvent } from './libs/enums/enums.js';
+import { type ServerSocketEventParameter } from './libs/types/types.js';
 
 class SocketService {
   private io: SocketServer | null = null;
 
+  private geolocationCacheService!: GeolocationCacheService;
+
   public getIo(): SocketServer {
     return this.io as SocketServer;
   }
 
-  public initializeIo(app: FastifyInstance): void {
+  public initializeIo({
+    app,
+    geolocationCacheService,
+  }: {
+    app: FastifyInstance;
+    geolocationCacheService: GeolocationCacheService;
+  }): void {
     this.io = new SocketServer(app.server, {
       cors: { origin: '*' },
     });
 
+    this.geolocationCacheService = geolocationCacheService;
```

Comment:
this.geolocationCacheService = GeolocationCacheService.getInstance();

Reply:
It's better to use dependency injection pattern and pass instance of the service as a parameter, I suppose.